### PR TITLE
add negative index handling to slice method (fixes #32)

### LINKED ISSUE #32: slice doesn't work with negative parametres
This is an incompatibility with Buffer.slice. Should be an easy fix.

### PR DESCRIPTION
**Pull request by nicolashenry**
add negative index handling to slice method (fixes #32)

### CODE REVIEW

File: bl.js [hunk 1]
```diff
@@ -107,6 +107,12 @@ BufferList.prototype.get = function get (index) {
 
 
 BufferList.prototype.slice = function slice (start, end) {
+  if (typeof start === 'number' && start < 0) {
```
Comment (rvagg) [inline bl.js]: don't need braces for simple single-line `if` statements. `==` is enough for the typeof check too

**Comment by rvagg** [comment]
just to set my mind at ease would you mind throwing in tests that mirror the negative index examples in the Buffer API docs @ https://nodejs.org/api/buffer.html#buffer_buf_slice_start_end please?

File: bl.js (hunk 1 above)
Comment (nicolashenry) [inline bl.js]: I think "===" is supposed to be slightly faster but ok.

**Comment by mcollina** [comment]
LGTM

**Pull request merged**